### PR TITLE
Poor Aim now gives you Stormtrooper Aim.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

When shooting a gun with Poor Aim, you will now shoot anything but your target within orange(2).

## Why It's Good For The Game

The bonus spread isn't as effective and still allows you to hit your target if you're firing from certain angles. This method properly emulates stormtrooper aim.

The blinding on firing with poor aim doesn't work well under high lag situation as a tick can end up being extremely long, and additionally feels very janky to the end-user.

This additionally makes Poor Aim impact other situations than mining, as now every gun ends up with stormtrooper aim regardless of the initial projectile deviation.

## Changelog
:cl:
balance: Poor Aim no longer blinds you, and now properly reflects the aiming capabilities of the galaxy's finest.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -320,6 +320,8 @@
 
 /obj/item/gun/proc/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
 	if(user)
+		if(HAS_TRAIT(user, TRAIT_POOR_AIM)) //nice shootin' tex
+			target = pick(orange(2, target))
```
Comment (ATH1909) [inline code/modules/projectiles/gun.dm:324]: what happens if this selects YOU/your tile?

**Review by coiax — COMMENTED** [review]
You should straight up change the name of the quirk to Stormtrooper Aim.

**Comment by necromanceranne** [comment]
> You should straight up change the name of the quirk to Stormtrooper Aim.

Maybe don't use the actual name of IP attached to Disney.

**Comment by FloranOtten** [comment]
Nukie Aim, then?

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/modules/projectiles/gun.dm:324]: Funniest shit I ever seen

Comment (Iamgoofball) [reply code/modules/projectiles/gun.dm:324]: Well, you do have really bad aim.

**Comment by maxymax13** [comment]
> 
> 
> > You should straight up change the name of the quirk to Stormtrooper Aim.
> 
> Maybe don't use the actual name of IP attached to Disney.

disney did not invent the name "stormtrooper"

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (Jared-Fogle) [reply code/modules/projectiles/gun.dm:324]: `orange` explicitly doesn't choose the tile of the target.

**Comment by Farquaar** [comment]
> 
> 
> > You should straight up change the name of the quirk to Stormtrooper Aim.
> 
> Maybe don't use the actual name of IP attached to Disney.

https://en.wikipedia.org/wiki/Stormtrooper

**Comment by necromanceranne** [comment]
> > > You should straight up change the name of the quirk to Stormtrooper Aim.
> > 
> > 
> > Maybe don't use the actual name of IP attached to Disney.
> 
> https://en.wikipedia.org/wiki/Stormtrooper

Fair argument.

**Comment by ilikesaying** [comment]
sturmtruppen can aim, stormtroopers can't

**Comment by NanoCats** [comment]
Question, what is "orange(2)", I'm not a real coder.


Also can you still hit what you're going for eventually with this? I feel like you should, it should just suck and miss _most_ of the time, otherwise you might as well just take pacifist lol. 

I'll take anything over the ungodly blindness that the previous fix entailed though, can't wait for this to get merged.

**Comment by Jared-Fogle** [comment]
> Question, what is "orange(2)", I'm not a real coder.

Picks a tile within a radius of 2, other than the one of the object passed--in this case, the target.

**Comment by NanoCats** [comment]
> > Question, what is "orange(2)", I'm not a real coder.
> 
> Picks a tile within a radius of 2, other than the one of the object passed--in this case, the target.

ah, thanks.

Although I would seriously suggest making it have like, a 10% chance to hit the intended target, even a crappy shot will hit their intended mark _occasionally_. 10% is just an example number by the way it can be whatever.

**Comment by ATH1909** [comment]
I don't particularly see a reason why this has to exclude specifically the target's tile from what your shot is aimed at. I feel like it'd lead to a weird meta where you'd always aim one tile off from your target to have a better chance of hitting them.

Also, how does this quirk interact with aiming near the edge of your screen instead of directly at your target?

**Comment by RaveRadbury** [comment]
I PR'ed the recent change that added blind(1) to Poor Aim.

I think that Goof's PR here is better than my idea.

Goof's point that blind(1) can become bad during variable lag is a very valid one, it is mainly this criticism that leads me to believe that his PR is better.

Thank you for your time.

**Comment by Jared-Fogle** [comment]
> Goof's point that blind(1) can become bad during variable lag is a very valid one, it is mainly this criticism that leads me to believe that his PR is better.

Do you like his idea because it doesn't lag, or because you think that it's a legitimately better mechanic?

If the problem is lag, make a fullscreen UI animation *in place* of the blind. That way it is completely independent of lag.

**Review by coiax — CHANGES_REQUESTED** [review]
Change the name of the quirk and trait to "Stormtrooper Aim".

**Comment by Jared-Fogle** [comment]
CC @RaveRadbury -- I want an answer on my question, if for nothing but out of curiosity.

**Comment by RaveRadbury** [comment]
>Do you like his idea because it doesn't lag, or because you think that it's a legitimately better mechanic?

>If the problem is lag, make a fullscreen UI animation in place of the blind. That way it is completely independent of lag.
> CC @RaveRadbury -- I want an answer on my question, if for nothing but out of curiosity.

Part of the balance of the blind was that your target had a chance of getting away from you during it. With an animation that runs independent of game speed I would think that there would be some cases in which the game would be running so slowly that it would most definitely not provide this disadvantage.

Is your recommendation something that could be stacked on top of the suggested mechanic here? Absolutely. Stormtrooper aim is a return to form for the poor aim quirk, but with a better promise of not being free quirk points. 

If you have found the idea of blinking while shooting a fun addition to the immersion of the quirk, I would be interested in making an attempt at the method you have suggested. I must admit, the idea of making an animation from scratch gives me pause, even if it's something as simple as two horizontal bars closing in and receding. Thankfully we have a community to provide guidance and feedback on it.

**Comment by Jared-Fogle** [comment]
> Is your recommendation something that could be stacked on top of the suggested mechanic here? Absolutely. Stormtrooper aim is a return to form for the poor aim quirk, but with a better promise of not being free quirk points.

I just wanted to know if you're accepting this as a compromise or because you believe it's legitimately better. 😛

**Comment by NanoCats** [comment]
Unsure if my opinion matters to anyone, but I really just enjoyed my character not being able to shoot straight to save her life myself, and the blink change made me just sigh and drop it. Like, keep in mind this isn't a powergame thing,  I had several spare positive quirk points, I literally just took it for characterization.  It also messed me up enough as it was with the spread and I died a couple times in scenarios where I was forced to take up arms.  Like, how bad does the de-perk need to be given that it's optional? If somebody is really concerned about balance couldn't they just make an adjustment to how many quirk points it's worth? 

I also had to drop clown fan, but like, that doesn't do a hell of a lot, but it has made me a little sad.


I'm not sure how it would be if the blink change was implemented, like, would you have a ton of spread AND the blink animation?

Also ATH1909's comment is valid, I have been trained to aim not at my target, but "Beyond" my target... I'm not sure why I do this tbh.

I almost wish there was a _good_ aim positive perk on the opposite end of the spectrum but that sounds like something powergamey that people would take just because it's a clear cut advantage. I'm not sure what the fuck it would do given that guns are generally pretty accurate though.

**Comment by Jared-Fogle** [comment]
> I'm not sure how it would be if the blink change was implemented, like, would you have a ton of spread AND the blink animation?

Would be up to Rave if they PR'd it, I don't care one way or the other.

**Comment by coiax** [comment]
I'm planning to merge this as soon as Goofball actually changes the name of the trait and quirk to Stormtrooper Aim.

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (Grunnyyy) [reply code/modules/projectiles/gun.dm:324]: Just click your own tile yourself

**Comment by Iamgoofball** [comment]
ah damn it it conflicts

**Comment by Iamgoofball** [comment]
@coiax done

**Pull request merged**